src/App.tsx

```diff
@@ -1,14 +1,58 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as todosActions } from './features/todos';
 
 export const App: React.FC = () => {
+  const [loader, setLoader] = useState(false);
+  const dispatch = useAppDispatch();
+  const getTodosFromRedux = useAppSelector(state => state.todos);
+  const getModal = useAppSelector(state => state.currentTodo);
+
+  const handleLoadTodos = useCallback(() => {
+    setLoader(true);
+    getTodos()
+      .then((todosFromServer:Todo[]) => {
+        dispatch(todosActions.setTodos(todosFromServer));
+      })
+      .finally(() => {
+        setLoader(false);
+      });
+  }, [dispatch]);
+
+  useEffect(() => {
+    handleLoadTodos();
+  }, [handleLoadTodos]);
+
+  const getfilter = useAppSelector(state => state.filter);
+
+  const visibleTodos = () => {
+    let visTodos = getTodosFromRedux;
+
+    if (getfilter.query.length !== 0) {
+      visTodos = visTodos.filter((todo:Todo) => todo.title.toLowerCase().includes(getfilter.query.toLowerCase()));
+    }
+
+    if (getfilter.filter === 'completed') {
```

Comment:
move todo statuses to enum

Reply:
@SanyaBratashchuk can You explain this one in more detail? Todo statuses are defined by Todo type, which is already given in this task.